engine/src/main/java/org/camunda/bpm/engine/impl/db/EnginePersistenceLogger.java

```diff
@@ -314,9 +315,19 @@ public ProcessEngineException illegalValueForUserIdException(String id, String e
       exceptionMessage("028", "Illegal value '{}' for userId for GLOBAL authorization. Must be '{}'", id, expected));
   }
 
-  public AuthorizationException requiredCamundaAdminException() {
-    return new AuthorizationException(
-      exceptionMessage("029", "Required admin authenticated group or user."));
+  public AuthorizationException requiredCamundaAdminOrPermissionException(List<MissingAuthorization> missingAuthorizations) {
+    String exceptionCode;
+    StringBuilder sb = new StringBuilder();
+    sb.append("Required admin authenticated group or user");
+    if(missingAuthorizations != null && !missingAuthorizations.isEmpty()) {
+      sb.append(" or any of the following permissions: ");
+      sb.append(AuthorizationException.generateMissingAuthorizationsList(missingAuthorizations));
+      exceptionCode = "110";
+    } else {
+      exceptionCode = "029";
+    }
```

Comment:
See previous suggestion for code initialization.
```suggestion
    }
```